cmd/mavenBuild.go

```diff
@@ -39,6 +40,53 @@ func mavenBuild(config mavenBuildOptions, telemetryData *telemetry.CustomData, c
 	}
 }
 
+func executeMavenGoals(config *mavenBuildOptions, utils maven.Utils, flags []string, goals []string, defines []string) error {
+	mavenOptions := maven.ExecuteOptions{
+		Flags:                       flags,
+		Goals:                       goals,
+		Defines:                     defines,
+		PomPath:                     config.PomPath,
+		ProjectSettingsFile:         config.ProjectSettingsFile,
+		GlobalSettingsFile:          config.GlobalSettingsFile,
+		M2Path:                      config.M2Path,
+		LogSuccessfulMavenTransfers: config.LogSuccessfulMavenTransfers,
+	}
+
+	_, err := maven.Execute(&mavenOptions, utils)
+	return err
+}
+
+func runMakeBOMGoal(config *mavenBuildOptions, utils maven.Utils) error {
+	var flags = []string{"-update-snapshots", "--batch-mode"}
+	if len(config.Profiles) > 0 {
+		flags = append(flags, "--activate-profiles", strings.Join(config.Profiles, ","))
+	}
+	exists, _ := utils.FileExists("integration-tests/pom.xml")
+	if exists {
+		flags = append(flags, "-pl", "!integration-tests")
+	}
+
+	var defines []string
+
+	createBOMConfig := []string{
+		"-DschemaVersion=1.4",
+		"-DincludeBomSerialNumber=true",
+		"-DincludeCompileScope=true",
+		"-DincludeProvidedScope=true",
+		"-DincludeRuntimeScope=true",
+		"-DincludeSystemScope=true",
+		"-DincludeTestScope=false",
+		"-DincludeLicenseText=false",
+		"-DoutputFormat=xml",
+		"-DoutputName=" + mvnSimpleBomFilename,
+	}
+	defines = append(defines, createBOMConfig...)
```

Comment:
I suppose we can rename `createBOMConfig` to `defines` and avoid creating a copy of the same slice?